hardware/areca.py

```diff
@@ -94,60 +96,98 @@ def run_areca(*args):
     return proc.communicate()[0]
 
 
-def run_and_parse(*args):
-    '''Run the areca command in a subprocess.
+def _run_and_parse(*args, rev=False):
+    """Run the areca command in a subprocess.
 
-Returns the output as an associative array.
-'''
-    res = run_areca(*args)
-    return parse_output(res)
+    Returns the output as an associative array.
+    """
+    res = _run_areca(*args)
+    return _parse_output(res, rev=rev)
 
 
-def run_and_parse_rev(*args):
-    '''Run the areca command in a subprocess.
-
-Returns the output as an associative array.
-'''
-    res = run_areca(*args)
-    return parse_output(res, rev=True)
-
-
-def adsys_info():
-    'Get advanced system info.'
-    arr = run_and_parse('adsys info')
+def _adsys_info():
+    """Get advanced system info."""
+    arr = _run_and_parse('adsys info')
     return arr
 
 
-def hw_info():
-    'Get hardware info.'
-    arr = run_and_parse_rev('hw info')
+def _hw_info():
+    """Get hardware info."""
+    arr = _run_and_parse('hw info', rev=True)
     return arr
 
 
-def sys_info():
-    'Get system info.'
-    arr = run_and_parse('sys info')
+def _sys_info():
+    """Get system info."""
+    arr = _run_and_parse('sys info')
     return arr
 
 
-def sys_showcfg():
-    'Get system configuration.'
-    arr = run_and_parse('sys showcfg')
+def _sys_showcfg():
+    """Get system configuration."""
+    arr = _run_and_parse('sys showcfg')
     return arr
 
 
-def hdd_pwr_info():
-    'Get hard disk power information.'
-    return run_and_parse('hddpwr info')
+def _hdd_pwr_info():
+    """Get hard disk power information."""
+    return _run_and_parse('hddpwr info')
 
 
-def disk_info(disk):
-    'Get hard disk information.'
-    output = run_and_parse('disk info drv=%d' % disk)
+def _disk_info(disk):
+    """Get hard disk information."""
+    output = _run_and_parse('disk info drv=%d' % disk)
     return output
 
 
-def disable_password():
-    'Command to temporarly disable password on the cli'
-    run_areca('set password=0000')
-# areca.py ends here
+def _disable_password():
+    """Command to temporarly disable password on the cli"""
+    _run_areca('set password=0000')
+
+
+def detect():
+    """Detect Areca controller configuration."""
+    hwlist = []
+    device = _sys_info()
+    if not device:
+        sys.stderr.write('Info: No Areca controller found\n')
+        return
+    if "ControllerName" not in device.keys():
+        sys.stderr.write('Info: Cannot find Areca controller name\n')
+        return
+
+    sys.stderr.write('Info: Areca controller found: %s version %s\n' %
+                     (device['ControllerName'], device['FirmwareVersion']))
+
+    _disable_password()
+
+    for info in device.keys():
```

Comment:
nit: we should transform iterating over keys into iterating over items

Reply:
yeah, I ended up ding this too